src/sidebar/components/TopBar.js

```diff
@@ -63,110 +64,101 @@ function TopBar({
-      )}
-      {auth.status === 'logged-out' && (
-        <span className="TopBar__login-links text-lg font-medium hyp-u-horizontal-spacing--2">
-          <LinkButton
-            classes="InlineLinkButton"
-            onClick={onSignUp}
-            style={loginLinkStyle}
-            variant="primary"
-          >
-            Sign up
-          </LinkButton>
-          <div>/</div>
-          <LinkButton
-            classes="InlineLinkButton"
-            onClick={onLogin}
-            style={loginLinkStyle}
-            variant="primary"
-          >
-            Log in
-          </LinkButton>
-        </span>
-      )}
-      {auth.status === 'logged-in' && (
-        <UserMenu auth={auth} onLogout={onLogout} />
-      )}
-    </>
-  );
-
   return (
-    <div className="TopBar">
-      {/* Single-annotation and stream views. */}
-      {!isSidebar && (
-        <div className="TopBar__inner content">
-          <StreamSearchInput />
-          <div className="hyp-u-stretch" />
-          <IconButton
-            icon="help"
-            expanded={isHelpPanelOpen}
-            onClick={requestHelp}
-            size="small"
-            title="Help"
-          />
-          {loginControl}
-        </div>
+    <div
+      className={classnames(
+        // This absolute-positioned top bar allows content to scroll beneath it;
+        // the use of `transform-gpu` can kick on hardware acceleration to make
+        // this smoother
+        'absolute h-10 left-0 top-0 right-0 z-4 transform-gpu',
+        'text-grey-7 border-b theme-clean:border-b-0 bg-white'
       )}
-      {/* Sidebar view */}
-      {isSidebar && (
-        <div className="TopBar__inner content">
-          <GroupList />
-          <div className="hyp-u-stretch" />
-          {pendingUpdateCount > 0 && (
-            <IconButton
-              icon="refresh"
-              onClick={applyPendingUpdates}
-              size="small"
-              variant="primary"
-              title={`Show ${pendingUpdateCount} new/updated ${
-                pendingUpdateCount === 1 ? 'annotation' : 'annotations'
-              }`}
-            />
-          )}
-          <SearchInput
-            query={filterQuery || null}
-            onSearch={store.setFilterQuery}
-          />
-          <SortMenu />
-          {showSharePageButton && (
-            <IconButton
-              icon="share"
-              expanded={isAnnotationsPanelOpen}
-              onClick={toggleSharePanel}
-              size="small"
-              title="Share annotations on this page"
-            />
+      data-testid="top-bar"
+    >
+      <SidebarContent
+        classes={classnames(
+          'flex items-center h-full',
+          // This precise horizontal padding makes the edges of its contents
+          // align accurately with the edges of annotation cards in the sidebar
+          'px-[9px]',
+          // Text sizing will size icons in buttons correctly
+          'text-xl'
+        )}
```

Comment:
In this case, we can more easily see how `TopBar` uses the shared `SidebarContent` pattern, but...also...this stuff.